cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/trigger/ProgramStatusTriggerBuilder.java

```diff
@@ -57,11 +47,8 @@ public ProgramStatusTriggerBuilder(@Nullable String programNamespace, @Nullable
   @Override
   public ProgramStatusTrigger build(String namespace, String applicationName, String applicationVersion) {
```

Comment:
Well the `build` method here just takes the namespace, applicationName, and applicationVersion all from the deployed environment. So how can it be a different appName? We only use the builder when the namespace, applicationName, and applicationVersion are all not specified.